flax/linen/recurrent.py

```diff
@@ -717,6 +744,12 @@ def scan_fn(
     else:
       carry, outputs = scan_output
 
+    if reverse and keep_order:
+      outputs = jax.tree_map(
```

Comment:
maybe factor this out into a separate function since you use it twice?

Reply:
Are we [DRY or WET](https://en.wikipedia.org/wiki/Don't_repeat_yourself)? For this use case I think abstracting away a tree_map leads to unnecessary indirection.